### ISSUE TITLE
Crash on irrefutable patterns

### ISSUE THREAD

**Issue by nikivazou** [ucsd-progsys/liquidhaskell#866]
The following code crashes Liquid Haskell

```
data Body = Body Int Int 

genBody :: Int -> Body
genBody s = Body x y 
  where 
    (x:y:_) = take 2 $ repeat s 
```

Error: 

```
 /Users/niki/liquidtypes/liquidhaskell/verified-instances/examples/nbody/allpairs.hs:4:1-7: Error: Uh oh.

 4 | genBody s = Body x y
     ^^^^^^^

     addC: malformed constraint:
{lq_tmp$x##263 : ({lq_tmp$x##259 : Int^"lq_tmp$x##248" | $k_##247[lq_tmp$x##245:=lq_anf$##1677723166##dpg][VV##246:=lq_tmp$x##259][VV##256:=lq_tmp$x##263]}, {lq_tmp$x##260 : Int^"lq_tmp$x##251" | $k_##250[VV##249:=lq_tmp$x##260][lq_tmp$x##245:=lq_anf$##1677723166##dpg][VV##256:=lq_tmp$x##263]})^"lq_tmp$x##258" "lq_tmp$x##2" | $k_##257[lq_tmp$x##245:=lq_anf$##1677723166##dpg][VV##256:=lq_tmp$x##263]}
 <:
Body((GHC.Types.Int, GHC.Types.Int), Main.Body)
```

**Comment by ranjitjhala** [comment]
This may be due to the simplifications done by eliminate .. can you run
with no simplify core?

**Comment by ranjitjhala** [comment]
fixed by #876
